airflow/providers/telegram/hooks/telegram.py

```diff
@@ -0,0 +1,154 @@
+    ) -> None:
+        super().__init__()
+        self.token = self.__get_token(token, telegram_conn_id)
+        self.chat_id = self.__get_chat_id(chat_id, telegram_conn_id)
+        self.connection = self.get_conn()
+
+    def get_conn(self) -> telegram.bot.Bot:
+        """
+        Returns the telegram bot client
+
+        :return: telegram bot client
+        :rtype: telegram.bot.Bot
+        """
+        return telegram.bot.Bot(token=self.token)
+
+    def __get_token(self, token: Optional[str], telegram_conn_id: str) -> str:
+        """
+        Returns the telegram API token
+
+        :param token: telegram API token
+        :type token: str
+        :param telegram_conn_id: telegram connection name
+        :type telegram_conn_id: str
+        :return: telegram API token
+        :rtype: str
+        """
+        if token is not None:
+            return token
+
+        if telegram_conn_id is not None:
+            conn = self.get_connection(telegram_conn_id)
+
+            if not conn.password:
+                raise AirflowException("Missing token(password) in Telegram connection")
+
+            return conn.password
+
+        raise AirflowException("Cannot get token: No valid Telegram connection supplied.")
+
+    def __get_chat_id(self, chat_id: Optional[str], telegram_conn_id: str) -> Optional[str]:
+        """
+        Returns the telegram chat ID for a chat/channel/group
+
+        :param chat_id: optional chat ID
+        :type chat_id: str
+        :param telegram_conn_id: telegram connection name
+        :type telegram_conn_id: str
+        :return: telegram chat ID
+        :rtype: str
+        """
+        if chat_id is not None:
+            return chat_id
+
+        if telegram_conn_id is not None:
+            conn = self.get_connection(telegram_conn_id)
+            return conn.host
+
+        return None
+
+    @tenacity.retry(
+        retry=tenacity.retry_if_exception_type(telegram.error.TelegramError),
+        stop=tenacity.stop_after_attempt(5),
+        wait=tenacity.wait_fixed(1),
+    )
+    def call(self, api_params: dict) -> None:
+        """
+        Sends the message to a telegram channel or chat.
+
+        :param method: not used
+        :type method: str
+        :param api_params: params for telegram_instance.send_message. It can also be used to override chat_id
+        :type api_params: dict
+        """
+        kwargs = {
+            "chat_id": self.chat_id,
+            "parse_mode": telegram.parsemode.ParseMode.HTML,
+            "disable_web_page_preview": True,
+        }
+        kwargs.update(api_params)
+
+        if 'text' not in kwargs or kwargs['text'] is None:
+            raise AirflowException("'text' must be provided for telegram message")
+
+        if kwargs['chat_id'] is None:
+            raise AirflowException("'chat_id' must be provided for telegram message")
+
+        response = self.connection.send_message(**kwargs)
+        self.log.debug(response)
```

Comment:
```suggestion
        self.log.info(response)
```
Do we need to log full response here?
Maybe log the message that call to API ends with success or other user friendly message. IMHO we should log here smth like `self.log.info("Message has been sent.")`

Reply:
I followed the pattern from other operators where we don't put anything in INFO mode unless very crucial. What do you suggest?

Reply:
I think I have never uses debug in my operators. I have no strong opinion here. I thought about confirmation and explicit info for the user that the operation ended with success. 
Like I mentioned in previous comment. 
But when I am looking at debug info today it make sense to log response. The question is if the DebugExecutor logs this  in the console, I supposed it is logged. Anyway, debug is ok for me.

Reply:
👍